### PR TITLE
Refactor module options to custom config table

### PR DESCRIPTION
**Pull request by rabauss**
For better review I splitted my changes for https://github.com/dklemmt/contao_dk_mmenu/pull/66

In this PR I refactor the module options to custom config table.
The translations get some optimization, too.

### CODE REVIEW

File: src/Migration/MmConfigMigration.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                                :dragOpenEnable,
+                                :dragOpenMaxStartPos,
+                                :dragOpenThreshold,
+                                :polyfillEnable,
+                                :keyboardNavigation,
+                                :keyboardNavigationEnhance
+                            )'
+                    );
+                    $stmt->executeQuery($config);
+                    $configId = (int) $this->connection->lastInsertId();
```
Comment (rabauss) [inline src/Migration/MmConfigMigration.php:175]: Is there a bette way to get the last config id of the insert?

File: src/Resources/contao/dca/tl_module.php [hunk 2]
```diff
@@ -16,266 +16,29 @@
...
-
-$GLOBALS['TL_DCA']['tl_module']['fields']['dk_mmenuOnClickClose'] = [
-    'label' => &$GLOBALS['TL_LANG']['tl_module']['dk_mmenuOnClickClose'],
-    'exclude' => true,
-    'inputType' => 'checkbox',
-    'eval' => ['tl_class' => 'clr w50'],
-    'sql' => "char(1) NOT NULL default ''",
-];
-
 $GLOBALS['TL_DCA']['tl_module']['fields']['dk_mmenuJsTpl'] = [
```
Comment (rabauss) [inline src/Resources/contao/dca/tl_module.php:25]: I decided to keep the dk_mmenuJsTpl in the module config - could also be moved to the config table but then the config loading cannot be handled in the `MmenuHelper`

File: src/Migration/MmConfigMigration.php (hunk 1 above)
Comment (fritzmg) [reply src/Migration/MmConfigMigration.php:175]: I don't think there is another way.

File: src/Resources/config/services.yml [hunk 3]
```diff
@@ -3,6 +3,11 @@ services:
         autowire: true
         autoconfigure: true
 
+    _instanceof:
+        Contao\CoreBundle\Framework\FrameworkAwareInterface:
```
Comment (rabauss) [inline src/Resources/config/services.yml:7]: TODO: refactor to DI with framework

**Comment by fritzmg** [comment]
@rabauss is this RTM?

**Comment by rabauss** [comment]
@fritzmg  Yes, this config feature should be ready + fixes the framework requirement of #68  !

**Comment by fritzmg** [comment]
Cool, merged :). Is there any additional work necessary? Otherwise I'd release `4.0.0`.

**Comment by rabauss** [comment]
@fritzmg maybe we could wait until the tests of mmenu v9 for `4.0.0`?

**Comment by fritzmg** [comment]
Alright. Are you actively testing? (I am not)

**Pull request merged**